rest-catalog/quarkus-rest-catalog/src/main/java/org/projectnessie/restcatalog/server/NessieApiProvider.java

```diff
@@ -47,10 +50,12 @@ public NessieApiV2 produceNessieApi() {
     return clientBuilder
         .fromSystemProperties()
         .fromConfig(cfg -> config.nessieClientConfig().get(cfg))
+        // FIXME: should we allow to configure the authentication method?
```

Comment:
I suspect, we'll need OAuth here, but this state is fine for me at the moment.

Reply:
Yes, the other way to do this is to configure an auth provider via config, for example the OAuth2 provider. I'm not sure it's better because this means that we'll have different credentials: one for Iceberg Client -> Iceberg REST Server communication, and another one for Iceberg REST Server -> Nessie Server. But I think we should let users decide, so I propose the following:

1. If an auth provider is provided via config, use it;
2. Otherwise, use the injected bean. By default, it should be `DynamicBearerAuthentication`, but if some other bean is present, use it.